### ISSUE TITLE
Typescript ts-loader error with sanity-content.d.ts

### ISSUE THREAD

**Issue by radum** [nuxt-community/sanity-module#68]
Hello,

Until I finish a demo project with this bug on the latest Nuxt version `"nuxt": "2.15.0",` and `"@nuxtjs/sanity": "^0.7.0",` has anyone seen this error:

```
ERROR in ./node_modules/@nuxtjs/sanity/dist/components/sanity-content.d.ts
Module build failed (from ./node_modules/ts-loader/index.js):
Error: Debug Failure. Output generation failed
    at Object.transpileModule (/Users/temp/packages/web/corporate/node_modules/typescript/lib/typescript.js:129814:29)
    at getTranspilationEmit (/Users/temp/packages/web/corporate/node_modules/ts-loader/dist/index.js:390:75)
    at successLoader (/Users/temp/packages/web/corporate/node_modules/ts-loader/dist/index.js:38:11)
    at Object.loader (/Users/temp/packages/web/corporate/node_modules/ts-loader/dist/index.js:23:5)
 @ ./node_modules/.cache/nuxt/components/plugin.js 9:11-11:5 16:9-18:3
 @ ./node_modules/.cache/nuxt/index.js
 @ ./node_modules/.cache/nuxt/client.js
 @ multi ./node_modules/@nuxt/components/lib/installComponents.js ./node_modules/.cache/nuxt/client.js

ERROR in ./node_modules/@nuxtjs/sanity/dist/components/sanity-image.d.ts
Module build failed (from ./node_modules/ts-loader/index.js):
Error: Debug Failure. Output generation failed
    at Object.transpileModule (/Users/temp/packages/web/corporate/node_modules/typescript/lib/typescript.js:129814:29)
    at getTranspilationEmit (/Users/temp/packages/web/corporate/node_modules/ts-loader/dist/index.js:390:75)
    at successLoader (/Users/temp/packages/web/corporate/node_modules/ts-loader/dist/index.js:38:11)
    at Object.loader (/Users/temp/packages/web/corporate/node_modules/ts-loader/dist/index.js:23:5)
 @ ./node_modules/.cache/nuxt/components/plugin.js 34:9-36:3 40:9-42:3
 @ ./node_modules/.cache/nuxt/index.js
 @ ./node_modules/.cache/nuxt/client.js
 @ multi ./node_modules/@nuxt/components/lib/installComponents.js ./node_modules/.cache/nuxt/client.js
```

package.json

```json
{
  "name": "@unilever/web-corporate",
  "version": "0.0.1",
  "private": true,
  "scripts": {
    "dev": "nuxt-ts -m=client",
    "build": "nuxt-ts build -m=client",
    "start": "nuxt-ts start -m=client",
    "generate": "nuxt-ts generate -m=client"
  },
  "dependencies": {
    "@nuxt/typescript-runtime": "^2.0.1",
    "@nuxtjs/axios": "^5.13.1s",
    "@nuxtjs/proxy": "^2.1.0",
    "@nuxtjs/pwa": "^3.3.5",
    "@nuxtjs/sanity": "^0.7.0",
    "@sanity/image-url": "^0.140.22",
    "axios-cache-adapter": "^2.7.0",
    "nuxt": "2.15.0",
    "nuxt-property-decorator": "^2.9.1",
    "nuxt-purgecss": "^1.0.0",
    "sanity-blocks-vue-component": "^0.1.0",
    "vue-lazy-hydration": "^2.0.0-beta.4",
    "vue-meta": "^2.4.0"
  },
  "devDependencies": {
    "@babel/runtime-corejs3": "^7.12.13",
    "@nuxt/types": "2.15.0",
    "@nuxt/typescript-build": "^2.0.4",
    "@nuxtjs/style-resources": "^1.0.0",
    "@nuxtjs/stylelint-module": "^4.0.0",
    "@testing-library/vue": "^5.6.1",
    "babel-core": "^7.0.0-bridge.0",
    "babel-jest": "^26.6.3",
    "core-js": "^3.8.3",
    "fibers": "^5.0.0",
    "jest": "^26.6.3",
    "jest-serializer-vue": "^2.0.2",
    "sass": "^1.32.7",
    "sass-loader": "10.1.1",
    "ts-jest": "^26.5.1",
    "vue-jest": "^4.0.0-rc.1"
  },
}
```

nuxt.config.js (for reference not the actual file)

```js
export default {
  target: 'static',

  plugins: [
    '~/plugins/sanity-image-builder.ts',
    '~/plugins/sanity-content.ts',
    '~/plugins/axios-hook-injection.server.ts',
    '~/plugins/preview.client.ts',
    '~/plugins/vue-directives.ts',
  ],

  components: [
    {
      path: '~/components',
      ignore: ['**/__mocks__/*.*', '**/__tests__/*.*']
    }
  ],

  buildModules: [
    '@nuxt/typescript-build',
    '@nuxtjs/stylelint-module',
    '@nuxtjs/sanity',
    '@nuxtjs/axios',
  ],

  modules: [
    '@nuxtjs/axios',
    '@nuxtjs/pwa',
    '@nuxtjs/style-resources',
  ],

  axios: {},

  proxy: {},

  content: {},

  build: {
    corejs: 3,
    extractCSS: true,
    optimizeCSS: true,
    splitChunks: {
      layouts: true,
      pages: true,
      commons: true,
    },
    optimization: {
      runtimeChunk: true,
    },
    babel: {
      presets({ envName }) {
        const envTargets = {
          client: { browsers: ['last 2 versions'], ie: 11 },
        };
        return [
          [
            '@nuxt/babel-preset-app',
            {
              targets: envTargets[envName],
              corejs: { version: 3 },
            },
          ],
        ];
      },
    },
    parallel: true,
    terser: false,
  },

  sanity: {
    projectId: process.env.SANITY_PROJECTID,
    dataset: process.env.SANITY_DATASET,
    token: process.env.SANITY_TOKEN,
    minimal: true,
    contentHelper: true,
    additionalClients: {
      previewClient: {
        withCredentials: true,
        useCdn: false,
      },
    },
  },
};
```

Works fine if I add `"@nuxtjs/sanity": "0.6.1",` back right before types were added.

**Comment by danielroe** [comment]
@radum Hopefully this should be fixed in 0.7.1. Let me know if it persists.

**Comment by radum** [comment]
Thanks @danielroe it works fine now.
